utils/genkeysymdef.js

```diff
@@ -108,7 +108,7 @@ out +=
 "};\n" +
 "\n" +
 "export default {\n" +
-"    lookup : function(u) {\n" +
+"    lookup(u) {\n" +
```

Comment:
Group this with other other function declaration cleanups in to a separate commit.

Reply:
What do you mean?
There are 2 commits:
1.- Change the objects to classes
2.- Change all the functions to arrow functions

(this change is part of the second one)

Reply:
I'd like a third one:

3.- Consistent function declarations (`foo: function()` => `function foo()`, `const foo = function()` => `function foo()`, etc.)

Reply:
I see. Currently:
1.- The shorthand notation is used if the function is a property of an object or class `{ foo() { ... } }` or `class bar { foo() { ... } }`
2.- The fat arrow notation `const foo = () => { ... };` for callback and any other function declared.
3.- The calssic `function foo() { ... }` for top level functions or functions that depend on the scope.

I prefer `const foo = () => { ... };` over `function foo() { ... }` for callbacks and functions that are not declared at the top level since it ensures that `foo` can't be overriden and automatically binds the function scope.

Would you like me to do a 3rd commit replacing all the `const foo = () => { ... };` by `function foo() { ... }` regardless of where they are?

Reply:
> So following this logic, lookup is correct as a shorthand function. It wouldn't even be a classic function unless we move it out of the object.

Fair enough. Might want to note that in the commit message in that case so that people aren't confused that the commit modifies more than just classes.

> I generally prefer const foo = () => { ... }; over function foo() { ... } for callbacks and functions that are not declared at the top level since it ensures that foo can't be overriden and automatically binds the function scope.
> 

What do you mean when you say it binds the function scope? Is there some practical detail I'm overlooking here?

I'm just concerned that `const foo = ()` is very similar to the `foo: function()` style we want to move away from. Also, coming from other languages, `const foo` reads more like "reference to an existing function" than a function definition, so that is tripping me up a bit.

Reply:
I meant that
`const foo = () => { ... };`
is equivalent to
```
function foo() { ... }
foo.bind(this)
```
so you don't need to do stuff like 
```
const self = this;
function foo() { ... <use self> ... }
```

Other difference is that const is safer to accidental overriding.
```
const foo = () => { ... };
foo = 2; // Throws an error saying that you can't override foo
foo();
```
while 
```
function foo() { ... }
foo = 2;
foo(); // Throws an error because foo is not a function
```

So nothing major. I think that my 3 points above summarize well my point of view :)